src/QirRuntime/lib/Simulators/ToffoliSimulator.cpp

```diff
@@ -67,21 +65,23 @@ namespace Quantum
         Qubit AllocateQubit() override
         {
             this->lastUsedId++;
-            this->states.ExtendToInclude(this->lastUsedId);
+            this->states.emplace_back();

```

Comment:
I have an impression that `this->states.emplace_back()` adds an *un*initialized bool to the vector.
If I'm mistaken then what is the value added to the vector?

Reply:
Actually, `emplace_back` will run the default constructor for the underlying type, which for `bool` means that the value will be initialized to false. I could make it explicit by instead using `emplace_back(false)` and that should avoid confusion for anyone working on this in the future.

Reply:
Actually, given your other comment and double checking the documentation, I'm now less sure that the default constructor for `bool` initializes to `false`. At the very least, it seems like it could vary by platform. I'll update this logic as described (use `emplace_back(false)`) and then review other places to ensure that I'm not assuming default value of false.